crates/sui-core/src/authority_server.rs

```diff
@@ -467,4 +470,14 @@ impl Validator for ValidatorService {
 
         return Ok(tonic::Response::new(response));
     }
+
+    async fn get_system_state_object(
+        &self,
+        _request: tonic::Request<SystemStateRequest>,
+    ) -> Result<tonic::Response<SuiSystemState>, tonic::Status> {
+        let epoch_store = self.state.load_epoch_store_one_call_per_task();
```

Comment:
this function's name is too scary. we should probably rename it as `load_epoch_store_unsafe_epoch_boundary`. In some cases, for example, we don't care about the inconsistency across epoch. Also it's very hard to make sure only one call per task, given our codebase size

Reply:
The name doesn't indicate that it should be called once per task, it just means that you should not hold on to the result any longer than necessary.

@lxfind should we require the client to specify the epoch number? Then we can detect if we have read the wrong version of the epoch store. But perhaps that's not necessary: the client can check if the epoch matches when they get the response.